tests/unit/test_client.py

```diff
@@ -11,29 +11,50 @@
 
 def test_urlopen_proxies(zap, urllib_mock):
     """Check if Zap client passes proxy to urllib call."""
-    urllib_mock.return_value.read.return_value = 'contents'
+    urllib_mock.return_value.text = 'contents'
 
     assert zap.urlopen() == 'contents'
     assert urllib_mock.mock_calls[0][2]['proxies'] == TEST_PROXIES
 
 
 def test_request_response(zap, urllib_mock):
     """Request method should return a python object from parsed output"""
-    urllib_mock.return_value.read.return_value = '{"testkey": "testvalue"}'
+    api_response = {"testkey": "testvalue"}
+    urllib_mock.api_response = api_response
+
+    assert zap._request('http://allizom.org', {'querykey': 'queryvalue'}) == api_response
+    assert zap._request_value('http://allizom.org', querykey='queryvalue') == 'testvalue'
 
-    assert zap._request('http://allizom.org', {'querykey': 'queryvalue'}) == {'testkey': 'testvalue'}
     assert urllib_mock.mock_calls == [
-        call('http://allizom.org?querykey=queryvalue', proxies=TEST_PROXIES),
-        call().read()
+        call(
+            'http://allizom.org',
+            params={'querykey': 'queryvalue', 'apikey': 'testapikey'},
+            proxies=TEST_PROXIES,
+            verify=False
+        ),
+        call().json(),
+        call(
+            'http://allizom.org',
+            params={'querykey': 'queryvalue', 'apikey': 'testapikey'},
+            proxies=TEST_PROXIES,
+            verify=False
+        ),
+        call().json()
     ]
 
 
 def test_request_other(zap, urllib_mock):
     """_request_other should simply return a retrieved content."""
-    urllib_mock.return_value.read.return_value = '{"testkey": "testvalue"}'
+    api_response = {"testkey": "testvalue"}
+    urllib_mock.api_response = api_response
 
-    assert zap._request('http://allizom.org', {'querykey': 'queryvalue'}) == {'testkey': 'testvalue'}
```

Comment:
I would be really good if we could also test if the key is set in the relevant header :)